pkg/internal/controllers/certificaterequests.go

```diff
@@ -238,22 +243,14 @@ func (c *certificaterequests) reconcileStatusPatch(ctx context.Context, req ctrl
 // Update the status with the provided condition details & return
 // the added condition.
 // NOTE: this code is just a workaround for apiutil only accepting the certificaterequest object
-func (c *certificaterequests) setCertificateRequestStatusCondition(
-	conditions *[]cmapi.CertificateRequestCondition,
+func (c *certificaterequests) getCertificateRequestStatusConditions(
```

Comment:
Could we use the https://github.com/cert-manager/issuer-lib/blob/main/conditions/certificaterequest.go implementation instead?

Reply:
We fixed these issues in that repo (see eg. https://github.com/cert-manager/issuer-lib/commit/aa4880ccbbe50d10992fb103aada6402694745b2).

Reply:
I'll give it a try.

Reply:
It's a new dependency, and I am not a big fan of dependencies. 😉 But if we think this new dependency is part of the future of this project it seems to work. Incoming commit.

Reply:
I stayed away from the latest version to avoid being forced to upgrade to Go 1.21.

Reply:
Please review and let me know what you think. I want to rebase before we merge something. Keeping the commits until we decide how to fix the issue.

Reply:
Sorry for the ambiguity, I was thinking we could copy the code from issuer-lib. I don't think we should have it as a dependency, since we are not building an issuer. What do you think?

Reply:
Ok, works for me. 😄 I want to get this issue fixed. After discovering the even earlier exit, I suggest to leave that duplication out and focus on the bug here: not taking `lastTranstionTime` from and existing condition into account. I wrote a test just for that - which is a bit pointless as the PR stands now. Since we exit before it becomes relevant.

Personally, I think the code in issuer-lib is too complex and hard to read. But I really want this bug fixed.

Reply:
Copied the function from issuer-lib now.